ethereumj-core/src/test/java/org/ethereum/core/ImportLightTest.java

```diff
@@ -372,15 +372,15 @@ public void createContractFork() throws Exception {
                 "}";
 
         StandaloneBlockchain bc = new StandaloneBlockchain();
-        SolidityContract parent = bc.submitNewContract(contractSrc, "Parent");
+        SolidityContract parent = bc.submitNewContract(contractSrc, "<stdin>:Parent");
```

Comment:
I think it's a bad thing with all this `<stdin>`'s.
We are making interface to proxy user from solc and add some simplicity and programming-language interface to interact with. Also all method declarations call it "Contract name" already. Plus we want it to work with several versions of solc.
I'd suggest modify CompilationResult.parse to remove "<stdin>:" part when it exists in the beginning and revert all method calls in tests. What do you think?

Reply:
Well, I didn't spent a lot of thoughts on this, simply adjusted the tests to work with the new Solidity compiler version.

To me it seems that the creators of Solidity, with version [0.4.9](https://github.com/ethereum/solidity/releases/tag/v0.4.9), added some form of a poor men's namespace support; that is, requiring to prefix each contract with `${filename}:`.  Only if the contract is read from standard input instead, the prefix becomes `<stdin>:`. Please follow the link above for the release notes.

I don't see how the new contract's namespace information could be ignored in the production code. Thoughts/concerns?

Reply:
Ok, what I suggest:
1) Adding public String path into `ContractMetadata`. If there is anything on first line before `:`, it's saved there
2) `public Map<String, ContractMetadata> contracts` becomes `public Map<String, List<ContractMetadata>> contracts`

What do you think? 
We are braking backward compatibility but I see only weird ways to preserve it.
Anyway remembering path which is specific to filesystem to get contract from compilation is not a good thing, I think, and we need to get it over.

Reply:
I understand your suggestion, but I believe that would not be as designed by the creators of Solidity.

To me it seems, since Solidity version `0.4.9`, that the compiler returns a contract key, not contract name anymore, which will **always** be prefixed with `${filename}:` or `<stdin>:`, that's a breaking change at the side of the Solidity compiler. You **cannot** ditch this information and use the contract name instead of the contract key as the key in `Map<String, ContractMetadata>`. This breaking change was introduced to allow contracts of the same name but different location to coexists, e.g. map entries with keys `/foo/file1.sol:test`, `/foo/bar/file2.sol:test`.

We need to agree on this first before we can think about how to better address this change.

Reply:
Look, now contract name just contains more information: path (or `<stdin>`) and contract name. I just suggest to store it separately in our Java model. Contract name is not {filename}:realContractName. It's just new part of information, they've added in output in such way. You are not able to call one contract from another with such prefix, so it's not a part of contract name, yeah?

Reply:
What do you want to use as key in `Map<String, ContractMetadata>`, `${filename}:realContractName` or `realContractName`?

Reply:
I suggest modify it to `Map<String, List<ContractMetadata>>`
usually it will be list with 1 `ContractMetadata`, but when there are two or more, user will get both in list and differentiate them by public field `path` which we'll add to  `ContractMetadata`. 
Or even better `Map<String, Map<String, ContractMetadata>>` -> `ContractName: Path: ContractMetadata`. As `path` is not a part of ContractMetadata by meaning.

Reply:
`Map<String, LinkedMap<String, ContractMetadata>>` so it could be accessible in simplest way: `contracts.get("MyContract").getValue(0)` 
@mkalinin what do you think?

Reply:
I can see how `Map<String, Map<String, ContractMetadata>>` could possibly work, but at what price? That seems to be a massive incompatibility for no good reason. The Solidity compiler returns a contract key on purpose but we then use the contract name instead as the `key` in our `Map<String, ?>`, no matter of the `value` type.
We clearly have opposite opinions on this topic, you prefer to use the contract name as `key` so that the tests continue to work while I believe a `key` is a `key` is a unique `key`. Any chance that you can bring in someone else from the team to be the tiebreaker?

Reply:
Yeah, I've called @mkalinin for his opinion, he will reply later.
I don't like paths (or combined key as originally comes from solc) as a first key because paths are even different on different filesystems and we make life of all users more difficult for rare cases when someone needs to compile the same names at once. Let's see what Michael will say.

Reply:
Yeah, I noticed, our messages somehow overlapped. I can relate to the discontent towards `java.io.File`, so maybe `Map<URI, ?>` would actually better reflect reality.

Reply:
One more thing, doing `contracts.get("MyContract").getValue(0)` as suggested by @zilm13 above would allow keeping the current tests more or less the same, however any general provider of this functionality would still have to check the entries in `LinkedMap<String, ContractMetadata>` in order to make sure that he doesn't miss any contract. It just unnecessarily complicates the code IMHO.

Reply:
Both of you have thoughts meaningful to me. I agree with @zilm13 that contract name is just a name and it's incorrect to prefix it with contract source, cause that will definitely confuse the user. On the other hand @tglaeser is right pointing that `CompilationResult` contains solidity output where `key` in `contracts` map is not a contract name but its identifier. So, what would I do: hide `CompilationResult.contracts` and provide getters to access it. I.e. something like `getContract(String name)` for general usage which returns first matching contract with no matter for path, `getContract(String name, String path)` - this method requires additional work in path transformation depending on the OS (need to check how solc handles that). Also, this change requires additional getter which returns just the first contract from that map (I saw that this case is also can take place). That is my opinion guys and that solution combines both of your suggestions, hence it should work :)

Reply:
Thanks for the input. I'll go forward and make the necessary changes and update the pull request...